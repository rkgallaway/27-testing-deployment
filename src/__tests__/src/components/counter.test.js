import React from "react";
import renderer from "react-test-renderer";

import Counter from "../../../components/counter/counter";

describe("<Counter />", () => {
  it("is Alive", () => {
    let component = shallow(<Counter />);
    expect(component.find(".count").text()).toBe('0');
  });

  it('can ccount up', () =>{
    let app = mount(<Counter />);
    app.find('.up').simulate('click');
    expect(app.state('count')).toEqual(1);
    app.find('.up').simulate('click');
    expect(app.state('count')).toEqual(2);
  });

  it('can ccount down', () =>{
    let app = mount(<Counter />);
    app.find('.down').simulate('click');
    expect(app.state('count')).toEqual(-1);
    app.find('.down').simulate('click');
    expect(app.state('count')).toEqual(-2);
  });

  it('visually displays proper polarity ad value on the counter', () =>{
    let app = mount(<Counter />);
    //at 0
    expect(app.find('.count.positive').exists()).toBeFalsy();
    expect(app.find('.count.negative').exists()).toBeFalsy();
    // at 1
    app.find('.up').simulate('click');
    expect(app.find('.count.positive').exists()).toBeTruthy();
    expect(app.find('.count').text()).toBe('1');
    //back down to 0
    app.find('.down').simulate('click');
    expect(app.find('.count').text()).toBe('0');
    expect(app.find('.count.positive').exists()).toBeFalsy();
    expect(app.find('.count.negative').exists()).toBeFalsy();
    //down to -1
    app.find('.down').simulate('click');
    expect(app.find('.count.negative').exists()).toBeTruthy();
    expect(app.find('.count').text()).toBe('-1');
  });
});

describe('Counter /> Core Component (Snapshot Test)', () =>{
  it('renders right', () =>{
    const component = renderer.create(<Counter />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});