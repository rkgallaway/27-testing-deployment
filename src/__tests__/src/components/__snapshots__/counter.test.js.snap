// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter /> Core Component (Snapshot Test) renders right 1`] = `
<section
  className="counter"
>
  <a
    className="down"
    href="#"
    onClick={[Function]}
  >
    down
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="up"
    href="#"
    onClick={[Function]}
  >
    up
  </a>
</section>
`;
